compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1934,9 +1936,146 @@ public BType getSafeType(BType type, boolean liftError) {
         return memberTypes;
     }
 
+    public boolean hasImplicitInitialValue(BType type) {
+        switch (type.tag) {
+            case TypeTags.INT:
+            case TypeTags.BYTE:
+            case TypeTags.FLOAT:
+            case TypeTags.DECIMAL:
+            case TypeTags.STRING:
+            case TypeTags.BOOLEAN:
+            case TypeTags.NIL:
+            case TypeTags.TYPEDESC:
+            case TypeTags.JSON:
+            case TypeTags.MAP:
+            case TypeTags.NONE:
+            case TypeTags.STREAM:
+            case TypeTags.TABLE:
+            case TypeTags.XML:
+            case TypeTags.ANYDATA:
+            case TypeTags.ANY:
```

Comment:
can't we use `type.tag < TypeTags.INVOKABLE`?

Reply:
We can go til type.tag < TypeTags.RECORD, as for record we need to look at members.
Still listing these constants feel more readable to me.